### PR TITLE
Breath first search exploration

### PR DESCRIPTION
**Pull request by henneboy**
Adds a slightly janky breath first search exploration algorithm.
I continuously moves to the nearest unseen tile.
The jankyness stems from it colliding with walls, but to fix it, it reversed 0.5 meters and tries again until it is successful.

280 lines of c#

Coauthored by: @miniJmo

### CODE REVIEW

File: Assets/Scripts/Robot/IRobotController.cs [hunk 1]
```diff
@@ -144,7 +144,7 @@ public interface IRobotController
         /// </summary>
         /// <returns> a list of all the tags within tag reading range of the robot</returns>
         List<RelativeObject<EnvironmentTag>> ReadNearbyTags();
-
+
```
Comment (Puvikaran2001) [inline Assets/Scripts/Robot/IRobotController.cs:147]:
```suggestion

```
Remove the tab, and then this file has no changes

File: Assets/Scripts/HenrikExample.cs [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+            {
+                var val = random.Next(0, 1000000);
+                rand_numbers.Add(val);
+            }
+
+            var constraintName = "Global";
+            var robotConstraints = constraintsDict[constraintName];
+
+            // var buildingConfigList50 = new List<BuildingMapConfig>();
+            // var buildingConfigList75 = new List<BuildingMapConfig>();
```
Comment (Puvikaran2001) [inline Assets/Scripts/HenrikExample.cs:127]:
```suggestion

```

File: Assets/Scripts/HenrikExample.cs [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+            var buildingConfigList100 = new List<BuildingMapConfig>();
+            foreach (int val in rand_numbers)
+            {
+                // buildingConfigList50.Add(new BuildingMapConfig(val, widthInTiles: 50, heightInTiles: 50));
+                // buildingConfigList75.Add(new BuildingMapConfig(val, widthInTiles: 75, heightInTiles: 75));
+                buildingConfigList100.Add(new BuildingMapConfig(val, widthInTiles: 100, heightInTiles: 100));
+            }
```
Comment (Puvikaran2001) [inline Assets/Scripts/HenrikExample.cs:134]:
```suggestion
            {
                buildingConfigList100.Add(new BuildingMapConfig(val, widthInTiles: 100, heightInTiles: 100));
            }
```

File: Assets/Scripts/HenrikExample.cs [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+            var constraintIterator = 0;
+            var mapSizes = new List<int> { 50, 75, 100 };
+            var algorithms = new Dictionary<string, RobotSpawner.CreateAlgorithmDelegate>
+                {
+                    { "random", seed => new HenrikExplorationAlgorithm() }
+                    //{ "tnf", seed => new TnfExplorationAlgorithm(1, 10, seed) }
+                    // { "minotaur", seed => new MinotaurAlgorithm(robotConstraints, seed, 2) },
+                    // { "greed", seed => new GreedAlgorithm() }
```
Comment (Puvikaran2001) [inline Assets/Scripts/HenrikExample.cs:143]:
```suggestion
                    { "HenrikExplAlgo", seed => new HenrikExplorationAlgorithm() }
```

File: Assets/Scripts/HenrikExample.cs [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+                };
+            constraintIterator++;
+            // var buildingMaps = buildingConfigList50.Union(buildingConfigList75.Union(buildingConfigList100));
+            var buildingMaps = buildingConfigList100;
```
Comment (Puvikaran2001) [inline Assets/Scripts/HenrikExample.cs:147]:
```suggestion
            var buildingMaps = buildingConfigList100;
```

File: Assets/Scripts/HenrikExample.cs [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+                        //                                                  mapSpawner: generator => generator.GenerateMap(mapConfig),
+                        //                                                  robotSpawner: (buildingConfig, spawner) => spawner.SpawnRobotsAtPositions(
+                        //                                                      collisionMap: buildingConfig,
+                        //                                                      seed: 123,
+                        //                                                      numberOfRobots: robotCount,
+                        //                                                      spawnPositions: spawningPosList,
+                        //                                                      createAlgorithmDelegate: algorithm),
+                        //                                                  statisticsFileName: $"{algorithmName}-seed-{mapConfig.RandomSeed}-size-{size}-comms-{constraintName}-robots-{robotCount}-SpawnApart",
+                        //                                                  robotConstraints: robotConstraints)
+                        // );
```
Comment (Puvikaran2001) [inline Assets/Scripts/HenrikExample.cs:184]:
```suggestion

```
Remove the uncomment code

**Review by Puvikaran2001 — CHANGES_REQUESTED** [review]
Looks good 👍.

Maybe remove the uncomment code in HenrikExample.cs

File: Assets/Scripts/Robot/IRobotController.cs [hunk 7]
```diff
@@ -144,7 +144,6 @@ public interface IRobotController
         /// </summary>
         /// <returns> a list of all the tags within tag reading range of the robot</returns>
         List<RelativeObject<EnvironmentTag>> ReadNearbyTags();
```
Comment (henneboy) [inline Assets/Scripts/Robot/IRobotController.cs:146]:
```suggestion
        List<RelativeObject<EnvironmentTag>> ReadNearbyTags();

```

**Review by Puvikaran2001 — APPROVED** [review]

**Pull request merged**